src/node_nanomsg.cc

```diff
@@ -37,7 +39,11 @@ NAN_METHOD(Close) {
     rc = nn_close(s);
   } while (rc < 0 && errno == EINTR);
 
-  info.GetReturnValue().Set(Nan::New<Number>(rc));
+  v8::Local<v8::Number> ret = Nan::New<Number>(rc);
+  Local<Value> argv[] = { ret };
+
+  info.GetReturnValue().Set(ret);
+  callback->Call(1, argv);
```

Comment:
That's a good explanation for why this was not fixing the timeout issue on Travis